### ISSUE TITLE
Upgrade jquery.flot

### ISSUE THREAD

**Issue by reedy** [emoncms/emoncms#1568]
Following on from https://github.com/emoncms/emoncms/issues/1567

https://github.com/emoncms/emoncms/blob/master/Lib/flot/NEWS.md says 0.8.3, and 4.2.0 is the latest release - https://github.com/flot/flot/releases

**Comment by reedy** [comment]
It's unclear if 511e987438d94a9eb4a9be1873f118c13567ae0a made some sort of fork, or it was a partial upgrade of the files...

**Comment by reedy** [comment]
https://github.com/emoncms/emoncms/blob/master/Lib/flot/date.format.js doesn't even seem to actually be a flot file, just dumped in the Lib/flot dir and merged into a combined file?

**Comment by reedy** [comment]
The same seems to be the case for https://github.com/emoncms/emoncms/blob/master/Lib/flot/excanvas.js

date.format is merged into a combined file for flot... excanvas is not?

**Comment by TrystanLea** [comment]
I think date.format.js ended up in the flot dir early on because it was included in some of the flot examples that the emoncms flot graphing was based on. We'd need to ammend the location for all optional modules as well if we change.

**Comment by reedy** [comment]
Yeah. It'd make sense to have a place for "vendor" (ie 3rd party) code and "in house" code.

Generally, I wouldn't expect "in house" code to live in a folder called lib, when it mostly contains "vendor" code
